### ISSUE TITLE
referring to module-disambiguated type within template fails to build

### ISSUE THREAD

**Issue by tersec** [nim-lang/Nim#21221]
### Description

```nim
type M = object
template r() = discard default(e.M)
discard default(e.M)  # builds
r()                   # doesn't build
```

Replacing
```nim
template r() = discard default(e.M)
```

with
```nim
template r() = discard default(M)
```
also builds.

The same phenomenon occurs across modules (import module `foo` which exports type `M`, then write `template r() = discard default(foo.M)`), but this demonstrates it within a single module.

### Nim Version

```
Nim Compiler Version 1.9.1 [Linux: amd64]
Compiled at 2023-01-03
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 81b7f9108f139bbf237f3e121fdbe9ff32a7193f
active boot switches: -d:release
```

```
Nim Compiler Version 1.6.11 [Linux: amd64]
Compiled at 2023-01-02
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 7f90bcf5b4cbe5b7da534dd79bafbb4cafb313fa
active boot switches: -d:release
```

```
Nim Compiler Version 1.6.10 [Linux: amd64]
Compiled at 2022-11-23
Copyright (c) 2006-2021 by Andreas Rumpf

active boot switches: -d:release
```

### Current Output

```text
/tmp/e.nim(4, 2) template/generic instantiation of `r` from here
/tmp/e.nim(2, 31) Error: type mismatch
Expression: default(M)
  [1] M: M

Expected one of (first mismatch at position [#]):
[1] proc default[T](_: typedesc[T]): T
```

### Expected Output

```text
Builds successfully
```

**Comment by metagn** [comment]
Using `typeof(e.M)` also works

**Comment by metagn** [comment]
Was *just* fixed in #22577 lol, missed the test case

Same as #19865
